### ISSUE TITLE
AWS S3 overwrite mode bug?

### ISSUE THREAD

**Issue by nskrjabins** [awslabs/aws-data-wrangler#135]
Hi,

Maybe  I am missing something very obvious but for me  the overwrite mode in `wr.pandas.read_parquet()` is not overwriting but appending. I have followed the tutorial to overwrite from [here](https://github.com/awslabs/aws-data-wrangler/blob/master/tutorials/s3_write_modes.ipynb) and here is what I am getting:

The initial overwrite:

![1st](https://user-images.githubusercontent.com/60360098/74550434-4d852600-4f49-11ea-98bd-bfa69fab8278.png)

The subsequent overwrite is appending:

![2nd](https://user-images.githubusercontent.com/60360098/74550507-67bf0400-4f49-11ea-9580-0ef99830537b.png)

It is not the end of the world, as I can simply manually delete everything in the target directory but would be good to understand the cause if the issue.

Thanks!

**Comment by igorborgest** [comment]
Hi @nskrjabins,

I'm don't know what is happening in your environment yet. But I will analyse it on the weekend.

**Comment by igorborgest** [comment]
@nskrjabins I've tried several times to replicate your situation, but without success.

Even with this extreme situation running on SageMaker with low latency with S3 and overwriting 100 files everything worked like a charm for me.

![image](https://user-images.githubusercontent.com/2163472/74597566-3b39e380-5040-11ea-91bd-94f08fcd9bcc.png)

I also added this case of sequential overwrites in our tests.

**Comment by igorborgest** [comment]
@nskrjabins can you give more details about your environment?
Which version of wrangler are you using?
In which platform are you running?
Are you sure that you have all permissions related with list and delete objects on S3?

**Comment by nskrjabins** [comment]
@igorborgest:
**environment details:** - conda 4.8.1, Python 3.7.4, pandas 0.25.1, jupyter lab 1.1.4 (for IDE)
**Version of wrangler:** 0.3.0
**Platform:** macOS Catalina v10.15.2
**Permissions in S3:** Yes, I do have all permissions, as I can manually delete object in the same S3 path. Assuming that permission to delete objects manually also allows deleting them with wrangle.

**Comment by igorborgest** [comment]
@nskrjabins 

Could you please update your version to the new 0.3.2 and retry it?

**Comment by nskrjabins** [comment]
@igorborgest 

I've updated it to the 0.3.2 and retried it but still the same error.

**Comment by nskrjabins** [comment]
I've also tried running the same script from a new virtual environment with **Python 3.7.4** and just **awswrangler 0.3.2**, still had the same error.

**Comment by nskrjabins** [comment]
Ok - this is really weird. So I double-checked the permission for the bucket - I had both read and write permissions for my account. But I thought, well let's try to make the bucket **public** - giving read and write permissions to everyone. Once I did it, the overwrite method worked properly.

Now what is weird, is that when I switched I made the bucket private again - removing read and write permissions from everyone, the overwrite method still continued to work properly...
